### PR TITLE
Remove Application Metrics Specs

### PR DESCRIPTION
**Pull request by lindgrenj6**
This is largely up for discussion, these specs basically just spin up a promtheus exporter and make sure it's exporting the right metrics when we record things in the counter. They consistently cause inconsistent spec failures due to the server in one spec not being tore down in time to run the other.

I can see why these were written since they're making sure our wrapper metric class properly records the metrics - but we could potentially move that to a mock and make sure the `record_process` metric is getting called. 

-----

TL;DR: These specs cause intermittent test failures, maybe we should remove them since all they're testing is the prometheus exporter gem.

### CODE REVIEW

**Comment by slemrmartin** [comment]
@lindgrenj6 I know how ambiguous is testing of prometheus specs. I agree we don't need to test that prometheus works, but - as you wrote - instead of removing whole spec we can mock some parts and test our class only

**Comment by lindgrenj6** [comment]
@slemrmartin I added a spec in the operations worker - doesn't look like we have a spec for the collector metrics, but I don't think we expose anything there. The error on processing is about all we expose right now (for this worker).

**Review by syncrou — APPROVED** [review]
LGTM 👍

**Review by slemrmartin — APPROVED** [review]
